exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/mergereceiver/MergingRecordBatch.java

```diff
@@ -838,4 +838,13 @@ public void dump() {
         container, outgoingPosition, Arrays.toString(incomingBatches), Arrays.toString(batchOffsets),
         Arrays.toString(tempBatchHolder), Arrays.toString(inputCounts), Arrays.toString(outputCounts));
   }
+
+  @Override
+  public WritableBatch getWritableBatch(int startIndex, int length) {
+    VectorContainer partialContainer = new VectorContainer(context.getAllocator(), getSchema());
```

Comment:
Looks like getWritableBatch in MergingRecordBatch, SpilledRecordBatch, AbstractRecordBatch and RecordBatchLoader is same.  Is there any way to write a common function in the hierarchy and use it in all these scenarios?